src/components/organisms/SchedulePageModal/SchedulePageModal.tsx

```diff
@@ -1,162 +1,126 @@
-import React, { useState, useMemo, FC } from "react";
-import { startOfDay, addDays, isWithinInterval, endOfDay } from "date-fns";
+import React, {
+  useState,
+  useMemo,
+  FC,
+  useEffect,
+  useCallback,
+  MouseEventHandler,
+} from "react";
+import { startOfDay } from "date-fns";
 import { range } from "lodash";
+import classNames from "classnames";
 
-import { AnyVenue, VenueEvent } from "types/venues";
+import { Room } from "types/rooms";
 
-import { formatDate, formatDateToWeekday } from "utils/time";
-import { WithId, WithVenueId } from "utils/id";
-import { itemsToObjectByIdReducer } from "utils/reducers";
+import { formatDate } from "utils/time";
 import { isEventLiveOrFuture } from "utils/event";
 
 import { useConnectRelatedVenues } from "hooks/useConnectRelatedVenues";
 import { useVenueId } from "hooks/useVenueId";
+import { useUser } from "hooks/useUser";
 
-import { EventDisplay } from "components/molecules/EventDisplay/EventDisplay";
+import { Schedule } from "components/molecules/Schedule";
+import { ScheduleVenueDescription } from "components/molecules/ScheduleVenueDescription";
 
-type DatedEvents = Array<{
-  dateDay: Date;
-  events: Array<WithVenueId<VenueEvent>>;
-}>;
+import { ScheduleDay } from "components/molecules/Schedule/Schedule.types";
+
+import { scheduleDayBuilder } from "./SchedulePageModal.utils";
+
+import "./SchedulePageModal.scss";
 
 const DAYS_AHEAD = 7;
 
 interface SchedulePageModalProps {
   isVisible?: boolean;
 }
 
+export const emptyPersonalizedSchedule = {};
+
 export const SchedulePageModal: FC<SchedulePageModalProps> = ({
   isVisible,
 }) => {
   const venueId = useVenueId();
-  const {
-    parentVenue,
-    currentVenue,
-    relatedVenues,
-    relatedVenueEvents,
-  } = useConnectRelatedVenues({
+  const { userWithId } = useUser();
+  const userEventIds =
+    userWithId?.myPersonalizedSchedule ?? emptyPersonalizedSchedule;
+
+  const { relatedVenueEvents, relatedVenues } = useConnectRelatedVenues({
     venueId,
     withEvents: true,
   });
 
-  const relatedVenuesById: Partial<
-    Record<string, WithId<AnyVenue>>
-  > = relatedVenues.reduce(itemsToObjectByIdReducer, {});
+  const [relatedRooms, setRelatedRooms] = useState<Room[]>([]);
+
+  useEffect(() => {
+    const rooms: Room[] = [];
+    relatedVenues
+      .map((venue) => venue.rooms || [])
+      .forEach((venueRooms) => rooms.push(...venueRooms));
+    setRelatedRooms(rooms);
+  }, [relatedVenues]);
```

Comment:
Instead of using `.forEach` here and mutating the variable, we should be able to do this all at once with `.reduce` or `.flatMap` or similar:

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap

```suggestion
  useEffect(() => {
    const rooms: Room[] = relatedVenues.flatMap((venue) => venue.rooms || [])
    setRelatedRooms(rooms);
  }, [relatedVenues]);
```

Also, what's the benefit of doing this as a `useState` + `useEffect` instead of just a `useMemo` here?

Reply:
cool! Have never met `flatMap`. used it here: https://github.com/sparkletown/sparkle/pull/1302/commits/6c79ed94ad0acdbf8061bba26c4e73f7912ed2de
and transformed `useEffect` to `useMemo`. Good point.

Reply:
cool! Have never met `flatMap`. used it here: https://github.com/sparkletown/sparkle/pull/1302/commits/6c79ed94ad0acdbf8061bba26c4e73f7912ed2de
and replaced `useEffect` with `useMemo`. Good point.

Reply:
`flatMap` is super neat. One of the concepts I learnt about deeply in my Scala/functional programming days.